packages/@ember/-internals/runtime/lib/system/core_object.js

```diff
@@ -256,6 +256,7 @@ class CoreObject {
       });
     }
 
+    registerDestructor(self, () => self.destroy(), true);
```

Comment:
Note that this is specifically an eager destructor to match the previous semantics. The `.destroy` method would always have gotten called **before** the downstream objects are marked as `isDestroying` and **before** the instances own `.willDestroy` is called.